coalib/bearlib/aspects/Smell.py

```diff
@@ -0,0 +1,236 @@
+from coalib.bearlib.aspects import Taste, Root
+
+
+@Root.subaspect
+class Smell:
+    """
+    This aspect detects `code smells` or `bad smells` in your code.
+
+    """
+    class docs:
+        definition = """
+        `Smells` are certain structures in the code that indicate violation of
+        fundamental design principles. They are usually not bugs; they are not
+        technically incorrect and do not currently prevent the program from
+        functioning.
+        """
+        importance_reason = """
+        Even though they are not necessarily bugs, code smells increase the risk
+        of bugs or failure in the future and may slow down development.
+        """
+
+
+@Smell.subaspect
+class ClassSmell:
+    """
+    This aspect detects `code smells` or `bad smells` related to classes'
+    definitions in your codebase.
+    """
+    class docs:
+        definition = """
+        Class-level code smells are simply code smells indicating poorly
+        defined classes (including too large classes or God object, data clump
+        feature envy etc...) in our source code.
+        """
+        importance_reason = """
+        These classes (the classes containing code smells) should be
+        refactored for better readability and maintainability of our source
+        code.
+        """
+
+
+@Smell.subaspect
+class MethodSmell:
+    """
+    This aspect detects `code smells` or `bad smells` related to methods'
+    and functions definitions in your codebase.
+    """
+    class docs:
+        definition = """
+        Method-level code smells are simply code smells indicating poorly
+        defined method and or functions (too long method or functions, or
+        functions with too many parameters) in our source code.
+        """
+        importance_reason = """
+        Make our functions and methods unambiguous(by reducing the number of
+        parameters, easy to read(by reducing the length of our methods and
+        functions) and debug.
+        """
+        fix_suggestions = """
+        A fix for this would simply consist of redefining the functions
+        (and or method), making them shorter and reducing the number of
+        parameters (maybe by creating more functions or using libraries).
+        """
+
+    max_method_length = Taste[int](
+        'Represents the max number of lines for a method or a function\'s'
+        'definition.',
+        (40,), default=40)
```

Comment:
Hmm... Like `method_name_style`? Which could be `'snake_case'` or `'lowerCamelCase'` or `'UpperCamelCase'`...